comp/otelcol/collector/impl/collector.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package collectorimpl provides the implementation of the collector component for OTel Agent
+package collectorimpl
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname"
+	corelog "github.com/DataDog/datadog-agent/comp/core/log"
+	collectorcontrib "github.com/DataDog/datadog-agent/comp/otelcol/collector-contrib/def"
+	collectordef "github.com/DataDog/datadog-agent/comp/otelcol/collector/def"
+	"github.com/DataDog/datadog-agent/comp/otelcol/logsagentpipeline"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/datadogexporter"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+type dependencies struct {
+	fx.In
+	// Lc specifies the fx lifecycle settings, used for appending startup
+	// and shutdown hooks.
+	Lc fx.Lifecycle
+
+	// Log specifies the logging component.
+	Log              corelog.Component
+	Provider         otelcol.ConfigProvider
+	CollectorContrib collectorcontrib.Component
+	Serializer       serializer.MetricSerializer
+	LogsAgent        optional.Option[logsagentpipeline.Component]
+	HostName         hostname.Component
+}
+
+type collector struct {
+	deps dependencies
+	set  otelcol.CollectorSettings
+	col  *otelcol.Collector
+}
+
+// New returns a new instance of the collector component.
+func New(deps dependencies) (collectordef.Component, error) {
+	set := otelcol.CollectorSettings{
+		BuildInfo: component.BuildInfo{
+			Version:     "0.0.1",
+			Command:     "otel-agent",
+			Description: "Datedog Agent OpenTelemetry Collector Distribution",
+		},
+		Factories: func() (otelcol.Factories, error) {
+			factories, err := deps.CollectorContrib.OTelComponentFactories()
+			if err != nil {
+				return otelcol.Factories{}, err
+			}
+			if v, ok := deps.LogsAgent.Get(); ok {
+				factories.Exporters[datadogexporter.Type] = datadogexporter.NewFactory(deps.Serializer, v, deps.HostName)
+			} else {
+				factories.Exporters[datadogexporter.Type] = datadogexporter.NewFactory(deps.Serializer, nil, deps.HostName)
+			}
+			return factories, nil
+		},
+		ConfigProvider: deps.Provider,
+	}
+	col, err := otelcol.NewCollector(set)
+	if err != nil {
+		return nil, err
+	}
+	c := &collector{
+		deps: deps,
+		set:  set,
+		col:  col,
+	}
+
+	deps.Lc.Append(fx.Hook{
+		OnStart: c.Start,
+		OnStop:  c.Stop,
+	})
+	return c, nil
+}
+
+func (c *collector) Start(context.Context) error {
+	go func() {
+		if err := c.col.Run(context.Background()); err != nil {
```

Comment:
Should it use the context argument?

Reply:
Context passed to Start is startCtx https://github.com/uber-go/fx/blob/2379ac906bbf063deb99033c995584234388fc98/app.go#L608-L611 , it timesout. 
We shouldn't use it in any component start.  See Other components 
logs agent https://github.com/DataDog/datadog-agent/blob/728165ea11b425945aaab6bdfbf75426791fc441/comp/logs/agent/agent.go#L158
forwarder 
https://github.com/DataDog/datadog-agent/blob/c4f5159f4ed2d5c25767dd92d978731a80a2e6ad/comp/forwarder/defaultforwarder/forwarder.go#L49